### PR TITLE
Fix unit tests for users

### LINKED ISSUE #95: Восстановить работу тестов

### PR DESCRIPTION
**Pull request by TS2Admin**
resolve #95

### CODE REVIEW

File: Backend/Core/Infrastructure.Business/Users/UserService.cs [hunk 1]
```diff
@@ -18,29 +18,30 @@ public UserService(IUserRepository<T> repository, IIdentityUserService identityU
 
         public IIdentityUserService IdentityUserService { get; }
 
-        public abstract string[] GetSpecificRoles();
+        public abstract Task<string[]> GetSpecificRoles();
 
         public async Task<T> Create(string firstName, string lastName, string phoneNumber)
```
Comment (fedorkin) [inline Backend/Core/Infrastructure.Business/Users/UserService.cs]: I think "Create" is not suitable name for this the method. We don't create new entity, we only assign some properties, like firstName, lastName, roles.

Comment (fedorkin) [inline Backend/Core/Infrastructure.Business/Users/UserService.cs]: We create entity by identity user. The key for creating is phone number. I think name "CreateByPhoneNumber" more suitable for this situation.

File: Backend/Core/Infrastructure.Http/Users/IdentityUserRepository.cs [hunk 2]
```diff
@@ -25,8 +25,8 @@ public IdentityUserRepository(IIdentityUsersAPI identityUsersAPI)
         public async Task Add(IdentityUser entity)
         {
             var userModel = Mapper.Map<UserModel>(entity);
-            var createUserModel = await IdentityUsersAPI.Create(userModel);
-            entity.Id = createUserModel.Id;
+            var createdUserModel = await IdentityUsersAPI.Create(userModel);
+            entity.Id = createdUserModel.Id;
```
Comment (fedorkin) [inline Backend/Core/Infrastructure.Http/Users/IdentityUserRepository.cs]: I don't know why, but I'm worried about this assigning. It's not an evident assigning for this method.

File: Backend/Core/UnitTests.Infrastructure.Business/Users/DriverServiceTests.cs [hunk 3]
```diff
@@ -41,249 +39,118 @@ public DriverServiceTestSuite()
...
 
         [Fact]
-        public async void CreateDriverWithExistIdentityUserResultCreatedDriver()
+        public async Task AssignVehicle()
         {
-            var suite = new DriverServiceTestSuite();
-
+            var driver = new Driver();
+            var vehicle = new Vehicle { Id = 3 };
             var company = new Company();
```
Comment (fedorkin) [inline Backend/Core/UnitTests.Infrastructure.Business/Users/DriverServiceTests.cs]: company isn't used in code.

File: Backend/Core/Infrastructure.Business/Users/UserService.cs (hunk 1 above)
Comment (fedorkin) [inline Backend/Core/Infrastructure.Business/Users/UserService.cs]: We create entity by identity user. The key for creating is phone number. I think name "CreateByIdentityPhoneNumber" more suitable for this situation.

File: Backend/Core/Infrastructure.Http/Users/IdentityUserRepository.cs (hunk 2 above)
Comment (fedorkin) [inline Backend/Core/Infrastructure.Http/Users/IdentityUserRepository.cs]: I don't know why, but I'm worried about this assigning. It's not an evident assigning for this method.

Comment (MazZilaaaa) [reply Backend/Core/Infrastructure.Http/Users/IdentityUserRepository.cs]: I have fixed it

File: Backend/Core/Infrastructure.Business/Users/UserService.cs (hunk 1 above)
Comment (MazZilaaaa) [reply Backend/Core/Infrastructure.Business/Users/UserService.cs]: The issue #98 has been created

**Pull request merged**